File: JS/assignment4/PublicApi.DTO.v1/Todo/TodoPriority.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace PublicApi.DTO.v1.Todo;

public class TodoPriority
{
    public Guid Id { get; set; }
        
    [MaxLength(128)]
    public string PriorityName { get; set; } = default!;

    public int PrioritySort { get; set; }
        
    public DateTime SyncDt { get; set; } = DateTime.UtcNow;
}

File: JS/assignment6/src/views/todoPriority.ts
import type { TodoPriorityRow } from "../models/todoPriority.js";

export interface TodoPriorityDto {
  id: string;
  priorityName: string | null;
  prioritySort: number;
  syncDt: string;
}

export function priorityView(row: TodoPriorityRow): TodoPriorityDto {
  return {
    id: row.id,
    priorityName: row.priorityName,
    prioritySort: row.prioritySort,
    syncDt: row.syncDt,
  };
}
